source/core/slang-performance-profiler.cpp

```diff
@@ -31,11 +31,14 @@ namespace Slang
         }
         virtual void getResult(StringBuilder& out) override
         {
-            for (auto func : data)
+            char buffer[512];
+            for (const auto& func : data)
             {
-                out << func.key << ": \t";
+                memset(buffer, 0, sizeof(buffer));
+                snprintf(buffer, sizeof(buffer), "[*] %30s", func.key);
```

Comment:
`memset` is unnecessary, but you may want to make sure that it doesn't cause a buffer overflow.
And 512 seems too big for outputting only 36 characters.
```
char buffer[40];
snprintf(buffer, sizeof(buffer), "[*] %30s", func.key);
buffer[39] = 0;
```

Reply:
We don't have a restriction on the length of `func.key` so I am using a large buffer just in case (so its not just 36 chars).

Reply:
Aha. Sorry. My mistake. I thought it was printing only the first 30 characters.